src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/action/start/TransportStartTransformAction.kt

```diff
@@ -0,0 +1,150 @@
+) {
+    private val log = LogManager.getLogger(javaClass)
+
+    override fun doExecute(task: Task, request: StartTransformRequest, actionListener: ActionListener<AcknowledgedResponse>) {
+        val getReq = GetTransformRequest(request.id(), null)
+        client.execute(GetTransformAction.INSTANCE, getReq, object : ActionListener<GetTransformResponse> {
+            override fun onResponse(response: GetTransformResponse) {
+                val transform = response.transform
+                if (transform == null) {
+                    return actionListener.onFailure(
+                        ElasticsearchStatusException("Could not find transform [${request.id()}]",
+                            RestStatus.NOT_FOUND)
+                    )
+                }
+
+                if (transform.enabled) {
+                    log.debug("Transform job is already enabled, checking if metadata needs to be updated")
+                    return if (transform.metadataId == null) {
+                        actionListener.onResponse(AcknowledgedResponse(true))
+                    } else {
+                        getTransformMetadata(transform, actionListener)
+                    }
+                }
+
+                updateTransformJob(transform, request, actionListener)
+            }
+
+            override fun onFailure(e: Exception) {
+                actionListener.onFailure(ExceptionsHelper.unwrapCause(e) as Exception)
+            }
+        })
+    }
+
+    private fun updateTransformJob(
+        transform: Transform,
+        request: StartTransformRequest,
+        actionListener: ActionListener<AcknowledgedResponse>
+    ) {
+        val now = Instant.now().toEpochMilli()
+        request.index(INDEX_MANAGEMENT_INDEX).doc(mapOf(Transform.TRANSFORM_TYPE to mapOf(Transform.ENABLED_FIELD to true,
+            Transform.ENABLED_AT_FIELD to now, Transform.UPDATED_AT_FIELD to now)))
+        client.update(request, object : ActionListener<UpdateResponse> {
+            override fun onResponse(response: UpdateResponse) {
+                if (response.result == DocWriteResponse.Result.UPDATED) {
+                    // If there is a metadata ID on transform then we need to set it back to STARTED or RETRY
+                    if (transform.metadataId != null) {
+                        getTransformMetadata(transform, actionListener)
+                    } else {
+                        actionListener.onResponse(AcknowledgedResponse(true))
+                    }
+                } else {
+                    actionListener.onResponse(AcknowledgedResponse(false))
+                }
+            }
+
+            override fun onFailure(e: Exception) {
+                actionListener.onFailure(ExceptionsHelper.unwrapCause(e) as Exception)
+            }
+        })
+    }
+
+    private fun getTransformMetadata(transform: Transform, actionListener: ActionListener<AcknowledgedResponse>) {
```

Comment:
This is a little confusing name, the method is actually attempting to fetch and updateTransformMetadata not just fetching transformMetadata.

Do you think it makes sense to rename this to something along lines of `retrieveAndUpdateTransformMetadata`?